supergloo/docs/mesh/tutorials/linkerd/tutorials-1-retries.md

```diff
@@ -0,0 +1,266 @@
+---
+title: "Tutorial: Configuring Retries"
+menuTitle: Retries
+description: Tutorial on how to configure Linkerd for Retries.
+weight: 3
+---
+
+### Summary
+
+In this tutorial we'll take a look at how to use SuperGloo and Linkerd to retry failed requests
+within our mesh.
+
+Prerequisites for this tutorial:
+
+- [SuperGloo Installed]({{% ref "/installation" %}})
+- [Linkerd Installed]({{% ref "/mesh/install-linkerd" %}})
+
+### Concepts
+
+**Retries**:
+
+Having a service mesh installed to your cluster allows you to automatically retry 
+failed requests. Not all requests should be retried - only requests that have an idempotent 
+result, such as most HTTP GET requests are safe to retry.
+
+By placing the task of retrying failed requests on the service mesh, your applications will no
+longer need to implement their own retry logic.
+
+This tutorial will show you how to set a retry policy to handle intermittent failures 
+using Linkerd as the base mesh.
+
+**RoutingRules**:
+
+Automatic Retries are achieved in SuperGloo with the use of *RoutingRules*. 
+
+For an in-depth overview of RoutingRules, see [the Istio Traffic Shifting Tutorial]({{% ref "/mesh/tutorials/istio/tutorials-5-security" %}})
+
+### Tutorial
+
+Now we'll demonstrate the RetryPolicy RoutingRule using a service with intermittent failures.
+
+First, ensure you've:
+
+- [installed SuperGloo]({{< ref "/installation" >}})
+- [installed Linkerd using SuperGloo]({{< ref "/mesh/install-linkerd" >}})
+
+Now we'll deploy 2 pods to our cluster:
+- a service that fails HTTP requests intermittently.
+- a pod running `sleep` that we can `kubectl exec` into and run commands.
+
+First, to ensure they'll get injected with the Linkerd sidecar:
+
+```bash
+kubectl annotate namespace default linkerd.io/inject=enabled
+```
+
+> Feel free to replace namespace `default` with one of your choosing.
+
+Now, deploy the services:
+
+```bash
+kubectl apply -n default -f \
+  https://raw.githubusercontent.com/solo-io/supergloo/master/test/e2e/files/test-service.yaml
+kubectl apply -n default -f \
+  https://raw.githubusercontent.com/solo-io/supergloo/master/test/e2e/files/testrunner.yaml
+```
+
+We should see the pods get created with the sidecar:
+
+
+```bash
+kubectl get pod -n default
+```
+
+```
+NAME                               READY   STATUS    RESTARTS   AGE
+test-service-v1-664b69bb58-4kmhs   2/2     Running   0          43s
+testrunner-77776ccd75-bqltg        2/2     Running   0          1m
+```
+
+We know the pods have had their sidecar injected if their ready count is `2/2`.
+
+Let's open a terminal and exec into the `testrunner` pod:
+
+```bash
+TESTRUNNER=$(kubectl get pod -n default -l app=testrunner -ojsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
```

Comment:
Think it would make sense to point a pointer here about using an Ingress like Gloo?

Reply:
in the case of Gloo, it'd be the ingress itself providing the retry feature rather than the mesh